src/modules/seasons/seasons.controller.ts

```diff
@@ -1,12 +1,46 @@
-import { Controller, Get } from '@nestjs/common';
+import { 
+    Body, 
+    Controller, 
+    Get, 
+    Param, 
+    Post, 
+    UsePipes, 
+    ValidationPipe 
+} from '@nestjs/common';
+import { NotFoundException } from '@nestjs/common/exceptions';
 import { SeasonsService } from './seasons.service';
+import { CreateSeasonDto, FindOneSeasonParam, SeasonDto } from './seasons.types';
 
 @Controller('seasons')
 export class SeasonsController {
     constructor(private readonly seasonsService: SeasonsService) {}
 
     @Get()
-    getSeasons(): object {
-        return this.seasonsService.findAll();
+    async getSeasons(): Promise<SeasonDto[]> {
+        const res = await this.seasonsService.findAll();
+        if (res)
+            return res;
+        else
+            throw new NotFoundException();
+    }
+
+    @Get(':id')
+    @UsePipes(new ValidationPipe())
+    async getOneSeason(@Param() params: FindOneSeasonParam): Promise<SeasonDto | object> {
+        const res = await this.seasonsService.findOne(params.id);
+        if (res)
+            return res;
+        else
+            throw new NotFoundException();
+    }
+
+    @Post()
+    @UsePipes(new ValidationPipe())
+    async createSeason(@Body() createSeasonDto: CreateSeasonDto): Promise<SeasonDto> {
+        const res = await this.seasonsService.createSeason(createSeasonDto);
+        if (res)
+            return res;
+        else
+            throw new NotFoundException();
```

Comment:
I don't think this can be a NotFoundException, because we're not looking for anything so I don't think we need this check/error. It would be very unlikely that this returns null since we're creating a new season so it should always return the entity.

More likely is we throw an `InternalServerError` from the `seasonService` method, and that gets returned to the requestor.